internal/service/timestreaminfluxdb/db_instance.go

```diff
@@ -0,0 +1,855 @@
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceDbInstance) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	conn := r.Meta().TimestreamInfluxDBClient(ctx)
+
+	var plan, state resourceDbInstanceData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Only fields without RequireReplace() will cause an update.
+	// Any other field changes will cause the resource to be destroyed and recreated.
+	// for aws_timestreaminfluxdb_db_instance this is tags, log_delivery_configuration, and
+	// db_parameter_group_identifier.
+	if !plan.DBParameterGroupIdentifier.Equal(state.DBParameterGroupIdentifier) ||
+		!plan.LogDeliveryConfiguration.Equal(state.LogDeliveryConfiguration) {
+
+		in := &timestreaminfluxdb.UpdateDbInstanceInput{
+			Identifier: aws.String(plan.ID.ValueString()),
+		}
+
+		if !plan.DBParameterGroupIdentifier.IsNull() && !plan.DBParameterGroupIdentifier.Equal(state.DBParameterGroupIdentifier) {
+			in.DbParameterGroupIdentifier = aws.String(plan.DBParameterGroupIdentifier.ValueString())
+		}
+
+		if !plan.LogDeliveryConfiguration.IsNull() && !plan.LogDeliveryConfiguration.Equal(state.LogDeliveryConfiguration) {
+			var tfList []logDeliveryConfigurationData
+			resp.Diagnostics.Append(plan.LogDeliveryConfiguration.ElementsAs(ctx, &tfList, false)...)
+			if resp.Diagnostics.HasError() {
+				return
+			}
+			in.LogDeliveryConfiguration = expandLogDeliveryConfiguration(tfList)
+		}
+
+		out, err := conn.UpdateDbInstance(ctx, in)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.TimestreamInfluxDB, create.ErrActionUpdating, ResNameDbInstance, plan.ID.String(), err),
+				err.Error(),
+			)
+			return
+		}
+		if out == nil || out.Id == nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.TimestreamInfluxDB, create.ErrActionUpdating, ResNameDbInstance, plan.ID.String(), nil),
+				errors.New("empty output").Error(),
+			)
+			return
+		}
+	}
+
+	updateTimeout := r.UpdateTimeout(ctx, plan.Timeouts)
+	_, err := waitDbInstanceUpdated(ctx, conn, plan.ID.ValueString(), updateTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.TimestreamInfluxDB, create.ErrActionWaitingForUpdate, ResNameDbInstance, plan.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	// Update status to current status
+	readOut, err := findDbInstanceByID(ctx, conn, plan.ID.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.TimestreamInfluxDB, create.ErrActionSetting, ResNameDbInstance, plan.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	// Setting computed attributes
+	plan.Status = flex.StringToFramework(ctx, (*string)(&readOut.Status))
```

Comment:
Computed attributes are usually updated above, around line 580, after calling the update method. However, the status always needs to be updated after any update and when tags are updated `UpdateDbInstance` cannot be used. Updating computed attributes here covers all cases.